src/util/hash.rs

```diff
@@ -474,6 +474,18 @@ impl <T: BitcoinHash> MerkleRoot for Vec<T> {
     }
 }
 
+/// Objects which are referred to by hash
+pub trait BitcoinHash {
+    /// Produces a Sha256dHash which can be used to refer to the object
+    fn bitcoin_hash(&self) -> Sha256dHash;
+}
+
+impl BitcoinHash for Vec<u8> {
+    #[inline]
+    fn bitcoin_hash(&self) -> Sha256dHash {
+        Sha256dHash::from_data(&self[..])
+    }
+}
```

Comment:
Is this used anywhere? It's pretty surprising that the "bitcoin hash" of a Vec<u8> does not include its length prefix.

Reply:
Used [here](https://github.com/rust-bitcoin/rust-bitcoin/blob/master/src/blockdata/constants.rs#L101) but doesn't seem to be needed?

Reply:
Like, the `merkle_root()` method calls `.bitcoin_hash()` on a `Vec<u8>`? Yeah, that should be made more explicit.

Reply:
Can you add a commit fixing `merkle_root` to use `Sha256dEncoder` and then fix this commit so it doesn't add useless code?

Reply:
Addressed in: ba84548c8e13a4fd83dde417970cd5f2e9e995c6

Not sure which commit you mean?